### PR TITLE
413: Add logic for creating and display excluded keywords

### PR DESCRIPTION
**Pull request by slants3**
This PR introduces the logic for excluded keywords.

We use the a psql specific pattern matching expression of `!~*` to remove grants with descriptions which include the specific pattern.

Keywords with `null` type default to `include`.

image

### CODE REVIEW

File: packages/client/src/views/Keywords.vue [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
-  <AddKeywordModal :showModal.sync="showAddKeywordModal"/>
+    <b-row>
+        <b-col><h3>Include results with:</h3></b-col>
+        <b-col class="d-flex justify-content-end">
+            <div>
+                <b-button variant="success" @click="openAddKeywordModal">Add</b-button>
+            </div>
+        </b-col>
+    </b-row>
+    <b-table sticky-header="600px" hover :items="includeKeywords" :fields="fields">
```
Comment (as1729) [inline packages/client/src/views/Keywords.vue:11]: Add `<hr />`

File: packages/server/src/db/index.js [hunk 2]
```diff
@@ -176,7 +176,18 @@ async function getAgencyCriteriaForAgency(agencyId) {
 
     return {
         eligibilityCodes: enabledECodes.map((c) => c.code),
-        keywords: keywords.map((c) => c.search_term),
```
Comment (as1729) [inline packages/server/src/db/index.js:179]: Can we add test coverage to this method?

File: packages/client/src/views/Keywords.vue (hunk 1 above)
Comment (as1729) [inline packages/client/src/views/Keywords.vue:11]: Can we add created_at column?

File: packages/client/src/views/Keywords.vue [hunk 3]
```diff
@@ -35,12 +50,31 @@ export default {
...
+        {
+          key: 'created_at',
+          thStyle: { width: '20%' },
+          formatter: (value) => {
+            const date = new Date(value);
+            const month = date.getMonth() + 1;
+            const day = date.getDate();
+            const year = date.getFullYear();
+
+            return `${month}/${day}/${year}`;
```
Comment (as1729) [inline packages/client/src/views/Keywords.vue:68]: Could be simplified to-
```js
return moment(value, 'YYYY-MM-DD').format('M/DD/YYYY');
```

In line 43 you'll need to add `import moment from 'moment';`

**Review by as1729 — COMMENTED** [review]
@slants3 the tests look good to me- can you add some screenshots of the current experience?

File: packages/client/src/views/Keywords.vue (hunk 3 above)
Comment (slants3) [reply packages/client/src/views/Keywords.vue:68]: Good call!

**Review by as1729 — APPROVED** [review]

**Review by as1729 — APPROVED** [review]

**Pull request merged**